### PR TITLE
WIP: Keycloak deployment

### PR DESCRIPTION
**Pull request by dgoldstein1**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

<br/><br/>

2. What **changes to custom.yaml** are required?

<br/><br/>

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

<br/><br/>

### CODE REVIEW

File: keycloak/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+![img](images/require_ssl.png)
+
+
+### Login 
+
+Get the edge external service 
+
+```bash
+kubectl get svc | grep edge
```
Comment (dontlaugh) [inline keycloak/README.md:108]: Can we generate the URL like before?

File: Makefile [hunk 2]
```diff
@@ -125,4 +125,12 @@ remove-observables:
...
\ No newline at end of file
+	cd spire && make custom-ca
+
+.PHONY: keycloak
+keycloak:
+	cd keycloak && make keycloak
+
+.PHONY: remove-keycloak
+remove-keycloak:
+	cd keycloak && make remove-keycloak
```
Comment (dontlaugh) [inline Makefile:136]: Add a newline, please

File: keycloak/realms/greymatter.json [hunk 3]
```diff
@@ -0,0 +1,2229 @@
+{
+  "id": "greymatter",
```
Comment (dontlaugh) [inline keycloak/realms/greymatter.json:2]: If this schema changes, we will need to know how to regenerate it.
This is a realm create-> export?

**Review by dontlaugh — COMMENTED** [review]
A few comments. Still need to run through the creation.

**Comment by chrisbsmith** [comment]
@dgoldstein1 , I don't think I'm a fan of including this in the helm-charts repo.  Specifically because we're trying to limit this to just Grey Matter and the observables stack.  @moosecanswim launched [a new repo](https://github.com/greymatter-io/platform-helm-charts) for non GM helm charts.  This may or may not be the best place for keycloak.

I'm open to discussion, but I think we should try to keep the Grey Matter helm charts repo specific for GM.

**Comment by dgoldstein1** [comment]
@chrisbsmith that looks like a better place. Closing this in favor of https://github.com/greymatter-io/platform-helm-charts/pull/1

**Comment by dontlaugh** [comment]
@dgoldstein1 delete the remote branch whenever you're ready

**Pull request closed without merging**